### PR TITLE
Task 1255975: Support “Go to Disassembly” in Call Stack Window for OpenDebugAD7 scenarios

### PR DESCRIPTION
**Pull request by gc46**
Before:
image

After:
image

### CODE REVIEW

**Comment by gc46** [comment]
To-Do:
- Check status of GetInfo() call
- Add in null checks?
- Use bstrAddress instead of bstrAddressAbsolute? Both have the same value assignment in GetInfo(). (See AD7MemoryAddress.cs, Line 157.)
- Clean up extraneous comments.

File: src/OpenDebugAD7/AD7DebugSession.cs [hunk 1]
```diff
@@ -1477,14 +1477,20 @@ protected override void HandleStackTraceRequestAsync(IRequestResponder<StackTrac
                             }
                         }
 
+                        IDebugCodeContext2 memoryAddress = null;
```
Comment (WardenGnaw) [inline src/OpenDebugAD7/AD7DebugSession.cs:1480]: You will want to move this into the `if (frame != null)` on line 1461

File: src/OpenDebugAD7/AD7DebugSession.cs [hunk 2]
```diff
@@ -1477,14 +1477,20 @@ protected override void HandleStackTraceRequestAsync(IRequestResponder<StackTrac
...
+                        frame.GetCodeContext(out memoryAddress);
+                        var contextInfo = new CONTEXT_INFO[1];
+                        memoryAddress.GetInfo(enum_CONTEXT_INFO_FIELDS.CIF_ALLFIELDS, contextInfo); // test -- need to delete; need to check if call is successful
```
Comment (gregg-miskelly) [inline src/OpenDebugAD7/AD7DebugSession.cs:1483]: You just want to ask for `enum_CONTEXT_INFO_FIELDS.CIF_ADDRES`

Comment (gregg-miskelly) [inline src/OpenDebugAD7/AD7DebugSession.cs:1483]: `memoryAddress?.` since you aren't checking for success

Comment (gregg-miskelly) [inline src/OpenDebugAD7/AD7DebugSession.cs:1483]: Should probably have:

```C#
string instructionPointerReference = null;
if (memoryAddress?.GetInfo(enum_CONTEXT_INFO_FIELDS.CIF_ADDRES, contextInfo ==  == HRConstants.S_OK &&
contextInfo[0].dwFields.HasFlag(enum_CONTEXT_INFO_FIELDS.CIF_ADDRESS))
{
    instructionPointerReference = contextInfo[0].bstrAddress;
}

...

{
...
    InstructionPointerReference = instructionPointerReference
```

Comment (WardenGnaw) [inline src/OpenDebugAD7/AD7DebugSession.cs:1483]: No need to grab all fields, you can just ask for the Address. You will want to validate that the contextInfo also set the flag that it has a value. E.g.
`contextInfo[0].dwFields.HasFlag(enum_CONTEXT_INFO_FIELDS.CIF_ADDRESSABSOLUTE)` before using `contextInfo[0].bstrAddressAbsolute`

**Review by WardenGnaw — CHANGES_REQUESTED** [review]

**Review by gregg-miskelly — APPROVED** [review]
Otherwise LGTM

**Review by gregg-miskelly — APPROVED** [review]

**Review by WardenGnaw — APPROVED** [review]

**Pull request merged**